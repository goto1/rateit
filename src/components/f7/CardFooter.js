import React from "react";
import PropTypes from "prop-types";

const CardFooter = ({ className, children }) => {
  const classNames = `card-footer ${className ? className : ""}`.trim();
  return (
    <div className={classNames}>
      {children}
    </div>
  );
};

CardFooter.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired
};

export default CardFooter;
