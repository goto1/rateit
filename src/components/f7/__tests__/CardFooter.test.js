import React from "react";
import CardFooter from "../CardFooter";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

describe("CardFooter component", () => {
  it("should render correctly", () => {
    const tree = renderer.create(<CardFooter>HelloWorld!</CardFooter>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render a span", () => {
    const wrapper = shallow(
      <CardFooter>
        <span>HelloWorld!</span>
      </CardFooter>
    );
    const actual = wrapper.find("span").at(0).html();
    const expected = "<span>HelloWorld!</span>";
    expect(actual).toBe(expected);
  });

  it("should render with a specified className", () => {
    const wrapper = shallow(
      <CardFooter className="custom">HelloWorld!</CardFooter>
    );
    const actual = wrapper.props().className;
    const expected = "card-footer custom";
    expect(actual).toBe(expected);
  });
});
