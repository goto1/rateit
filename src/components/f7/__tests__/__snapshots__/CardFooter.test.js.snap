// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardFooter component should render correctly 1`] = `


<div
  className="card-footer"
>
  HelloWorld!
</div>
`;
